R/create_bgc_mask.R

```diff
@@ -0,0 +1,98 @@
+#' Create cost no sample areas per bec zone
+#'
+#' Creates and applies a mask of the bec zone to the final cost layer
+#'
+#' @param vec_dir text string with folder location of base vector layers
+#' @param cost_masked A `SpatRast` of cost layer with high cost penalty and
+#'      exclusions applied.
+#' @param out_dir A `character` or path which points to input location
+#'      of . A default location and name are applied in line with standard workflow.
+#' @return path to the output directory where files are written (invisibly).
+#' @param ... Additional options passed on to `...` in [terra::writeRaster()]
+#' @inheritParams terra::writeRaster
+#' @export
+#'
+#' @examples
+#' \dontrun{
+#' create_bgc_mask <- function(vec_dir, cost_final, out_dir)
+#'}
+#'
+create_bgc_mask <- function(
+   vec_dir = fs::path(PEMprepr::read_fid()$dir_1010_vector$path_abs),
+   cost_masked,
+   out_dir = fs::path(PEMprepr::read_fid()$dir_201010_inputs$path_abs),
+   overwrite = TRUE,
+   ...){
+
+  if (!inherits(vec_dir, c("character"))) {
+    cli::cli_abort("{.var vec_dir} must be a SpatRaster or a path to a file")
+  }
+
+  if (inherits(cost_masked, c("character"))) {
+    cost <- terra::rast(cost_masked)
+  } else if (!inherits(cost_masked, c("SpatRaster"))) {
+    cli::cli_abort("{.var cost_masked} must be a SpatRaster or a path to a file")
+  }
+
+  if (!fs::dir_exists(out_dir)) {
+    fs::dir_create(out_dir, recurse = TRUE)
+    cli::cli_alert_warning(
+      "write out folder does not exist, creating at location {.var out_dir}"
+    )
+  }
+
+  if (!fs::file_exists(fs::path(vec_dir, "bec.gpkg"))) {
+    cli::cli_abort(
+      "bec.gpkg does not exist in {.var vec_dir}. Please check the function
+        create_base_vectors() ran correctly or add this manually"
+    )
+  } else {
+    bec <- sf::st_read(fs::path(vec_dir, "bec.gpkg"))
+
+    if (!fs::dir_exists(out_dir)) {
+      fs::dir_create(out_dir, recurse = TRUE)
+      cli::cli_alert_warning(
+        "write out folder does not exist, creating at location {.var out_dir}}"
+      )
+    }
```

Comment:
I think we can just run this - it creates the dir if it doesn't exist, and is just silent if it does already

```suggestion
  fs::dir_create(out_dir, recurse = TRUE)
```